pgjdbc/src/test/java/org/postgresql/replication/ReplicationConnectionTest.java

```diff
@@ -58,17 +58,12 @@ public void testIsValid() throws Exception {
   public void testConnectionNotValidWhenSessionTerminated() throws Exception {
     int backendId = ((PGConnection) replConnection).getBackendPID();
 
-    Connection sqlConnection = TestUtil.openDB();
-
-    Statement terminateStatement = sqlConnection.createStatement();
-    terminateStatement.execute("SELECT pg_terminate_backend(" + backendId + ")");
-    terminateStatement.close();
-    sqlConnection.close();
+    TestUtil.terminateBackend(backendId);
```

Comment:
Frankly speaking, static assumption on implicit database looks odd.
It would probably be nice to have "database" or "database type" parameter.

For instance, if we add three-way database tests, then the meaning of `terminateBackend(backendId)` becomes moot.

So I don't want to make static `terminateBackend(int)` easily accessible.

WDYT?

Reply:
Yes that'd be nicer longer term. Maybe by changing the signature to take a Connection, unwrapping it as a PGConnection, and then creating a new connection to that host / port with the super user creds. The assumption would be the same super user creds across everything being tested.

I'll take a look if that will work. Right now there's a bunch of overloads and the real implementation already takes a Connection parameter as the connection used to perform the termination action (not the one being terminated) so I had left things as unchanged as possible to avoid changing the signatures too much.

Reply:
Yeah, you are right, adding method to PGConnection would be overkill for this PR. Let's skip it.

Reply:
Ha! Too late I already did :)

Wasn't as bad as I thought and the end result is much cleaner. Rest of your comments re:Duration and try-with-resources are there too.

Reply:
🙀